airbyte-integrations/connectors/destination-gcs-data-lake/src/main/kotlin/io/airbyte/integrations/destination/gcs_data_lake/catalog/GcsCredentialsProvider.kt

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2025 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.destination.gcs_data_lake.catalog
+
+import com.google.auth.Credentials
+import com.google.auth.oauth2.GoogleCredentials
+import java.io.ByteArrayInputStream
+
+const val GCS_CREDENTIALS_JSON = "gcs-credentials-json"
+
+/**
+ * Custom credentials provider for Google Cloud that can be instantiated by Iceberg's
+ * GoogleAuthManager via reflection.
+ *
+ * This class allows us to pass service account credentials programmatically to Iceberg without
+ * relying on environment variables or file system paths.
+ *
+ * Similar to GlueCredentialsProvider in S3DataLake destination.
+ */
+class GcsCredentialsProvider private constructor(private val delegate: GoogleCredentials) :
+    Credentials() {
+
+    override fun getAuthenticationType(): String = delegate.authenticationType
```

Comment:
all of these functions just pass through to the delegate - why do we to instantiate this class at all? I.e. why not just have the `create` function just `return scopedCredentials directly, rather than `return GcsCredentialsProvider(scopedCredentials)`